pint-abi-gen/src/lib.rs

```diff
@@ -73,16 +73,16 @@ fn ty_from_pint_ty(ty: &TypeABI) -> syn::Type {
 /// Names for fields are emitted by pint with a `::` prefix.
 /// This function checks for the `::` prefix and strips it if it exists.
 fn strip_colons_prefix(name: &str) -> &str {
-    let mut split = name.split("::");
-    let first = split.next();
-    split.next().or(first).expect("name had unexpected format")
+    name.trim_start_matches("::")
 }
 
-/// Var names have the `::` prefix, and sometimes have `.` separators for
-/// flattened tuple fields. We strip the `::` prefix, and replace all `.`
+/// Var names have the `::` prefix, and sometimes have `.` or `@` or `::` separators for
+/// flattened tuple fields. We strip the `::` prefix, and replace all `.` or `@` or `::`
 /// occurrences with `_`.
 fn field_name_from_var_name(name: &str) -> String {
-    strip_colons_prefix(name).replace('.', "_")
+    strip_colons_prefix(name)
+        .replace(['.', '@'], "_")
```

Comment:
My only concern here is the reason we used `@` in the first place.  By replacing it with `_` you could introduce a name clash.

Reply:
Like if someone names a var `anon_v5`?

Reply:
Yep.

Reply:
We might just have to take the risk. It will only affect abi gen. Although it would suck if you are trying to use the abi of an already deployed contract that has this name collision. I mean you could just change the name because that won't change the address of the contract.

I'm not sure what else we can do because rust only allows `[a-zA-Z][a-ZA-Z0-9_]*` for vairable / fields / functions.

Reply:
Yeah, I'm just thinking using Rust identifiers for Pint identifiers might not be the best.  An API that uses strings for identifiers instead of actual Rust identifiers would be safer.

Reply:
It makes it impossible to use Rust types then which is kind of the whole point of this.

If it's just strings then you are back to guessing what the fields are. Basically you need to keep looking at the json manually

Reply:
@mitchmindtree before I merge this what are your thoughts on this?

Reply:
I'm not too concerned with this as the ABI-gen is just a thing to improve devex, and if a name conflict does happen the Rust compiler will just complain about an already declared method or field.